packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -312,6 +330,178 @@ export default async (req, res) => {
   })
 }
 
+async function listAndCountProductWithIsolatedProductModule(
+  req,
+  filterableFields,
+  listConfig
+) {
+  // TODO: Add support for fields/expands
+
+  const remoteQuery = req.scope.resolve("remoteQuery")
+
+  let salesChannelIdFilter = filterableFields.sales_channel_id
+  if (req.publishableApiKeyScopes?.sales_channel_ids.length) {
+    salesChannelIdFilter ??= req.publishableApiKeyScopes.sales_channel_ids
+  }
+
+  delete filterableFields.sales_channel_id
+
+  filterableFields["categories"] = {
+    $or: [
+      {
+        id: null,
+      },
+      {
+        ...(filterableFields.categories || {}),
+        // Store APIs are only allowed to query active and public categories
+        ...defaultStoreCategoryScope,
+      },
+    ],
+  }
+
+  // This is not the best way of handling cross filtering but for now I would say it is fine
+  if (salesChannelIdFilter) {
+    const salesChannelService = req.scope.resolve(
+      "salesChannelService"
+    ) as SalesChannelService
+
+    const productIdsInSalesChannel =
+      await salesChannelService.listProductIdsBySalesChannelIds(
+        salesChannelIdFilter
+      )
+
+    let filteredProductIds = productIdsInSalesChannel[salesChannelIdFilter]
+
+    if (filterableFields.id) {
+      filterableFields.id = Array.isArray(filterableFields.id)
+        ? filterableFields.id
+        : [filterableFields.id]
+
+      const salesChannelProductIdsSet = new Set(filteredProductIds)
+
+      filteredProductIds = filterableFields.id.filter((productId) =>
+        salesChannelProductIdsSet.has(productId)
+      )
+    }
+
+    filterableFields.id = filteredProductIds
+  }
+
+  const variables = {
+    filters: filterableFields,
+    order: listConfig.order,
+    skip: listConfig.skip,
+    take: listConfig.take,
+  }
+
+  // prettier-ignore
+  const args = `
+    filters: $filters,
+    order: $order,
+    skip: $skip, 
+    take: $take
+  `
+
+  const commonProperties = ["id", "created_at", "updated_at", "deleted_at"]
```

Comment:
comment: Maybe it's just me, but I like it better with repeated properties than this `.join( ... )` approach. I know this makes for more maintainable code, so fine to keep it as is 😅